### ISSUE TITLE
Refactoring a method with "Change Signature" adds unnecessary named argument labels when caller has in-order argument label

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#66600]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Refactoring-a-method-with-Change-Signat/10257968)._

---
[severity:It's more difficult to complete my work]
When using the Change Signature refactoring gesture to reorder parameters, call sites get rewritten to the new order.
When a call site has any named arguments, even though they are positional (e.g. only there for readability) and not neccessary, the final ordering will cause all arguments occurring after the first originally named argument to be named arguments.

This is not an issue when the call site does not have any named arguments. Thank you!

## Steps to reproduce
1. Use "Change Signature" tool to move the last parameter (named `value`) to be the first.

Signature before refactoring:
```csharp
    public void MethodToRefactor(bool useMatcher2, StringBuilder writer, Regex matcher1, Regex matcher2, string value)
```

Signature after refactoring:
```csharp
    public void MethodToRefactor(string value, bool useMatcher2, StringBuilder writer, Regex matcher1, Regex matcher2)
```

## Expected Behavior:
The call sites should detect the positional, unnecessary named arguments and not add named arguments to those that follow.
This maintains the code cleanliness and intent by using positional named arguments, e.g. for flag parameters.

Callsites before:
```csharp
        MethodToRefactor(false, writer, matcher1, matcher2, "Hello world!");
        MethodToRefactor(useMatcher2: true, writer, matcher1, matcher2, "Hello world!");
```

Callsites after:
```csharp
        MethodToRefactor("Hello world!", false, writer, matcher1, matcher2);
        MethodToRefactor("Hello world!", useMatcher2: true, writer, matcher1, matcher2);
```

## Actual Behavior:
The rewriter detects that any named argument is used, it adds named arguments to all arguments after it.

Callsites before:
```csharp
        MethodToRefactor(false, writer, matcher1, matcher2, "Hello world!");
        MethodToRefactor(useMatcher2: true, writer, matcher1, matcher2, "Hello world!");
```

Callsites after:
```csharp
        MethodToRefactor("Hello world!", false, writer, matcher1, matcher2);
        MethodToRefactor("Hello world!", useMatcher2: true, writer: writer, matcher1: matcher1, matcher2: matcher2);
```

While this refactoring is correct, it causes a lot of extra code being added all over the codebase (Imagine a method used > 100 times), making the code harder to read and maintain. Trying to manually undo the unnecessary named arguments is prohibitive.

## Original code
For reference:

```csharp
using System;
using System.Text;
using System.Text.RegularExpressions;

namespace SampleCode;

public class ChangeSignatureTest
{
    public void MethodToRefactor(bool useMatcher2, StringBuilder writer, Regex matcher1, Regex matcher2, string value)
    {
        _ = value ?? throw new ArgumentNullException(nameof(value));
        _ = writer ?? throw new ArgumentNullException(nameof(writer));
        _ = matcher1 ?? throw new ArgumentNullException(nameof(matcher1));
        _ = matcher2 ?? throw new ArgumentNullException(nameof(matcher2));

var matcher = useMatcher2 ? matcher2 : matcher1;
        if (matcher. IsMatch(value))
        {
            writer. AppendLine($"The text '{value}' does not match the selected pattern");
        }
        else
        {
            writer. AppendLine($"The text '{value}' does not match the selected pattern");
        }
    }

public void Callers()
    {
        var matcher1 = new Regex("abc");
        var matcher2 = new Regex("123");
        var writer = new StringBuilder();

MethodToRefactor(false, writer, matcher1, matcher2, "Hello world!");
        MethodToRefactor(useMatcher2: true, writer, matcher1, matcher2, "Hello world!");
    }
}
```

---
#### Feedback Bot on 1/19/2023, 07:18 PM: 

(private comment, text removed)
#### Feedback Bot on 1/20/2023, 06:53 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by joem-msft** [comment]
This was my original vs feedback item.
